File: webflux-demo-java/src/main/java/za/co/entelect/springforum/webfluxdemo/dragons/Dragon.java
package za.co.entelect.springforum.webfluxdemo.dragons;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.net.URI;

@Data
@AllArgsConstructor
@Table("dragons")
public class Dragon {
    @Id
    @JsonIgnore
    private Integer id;

    private final String name;
    private final String location;

    URI uri() {
        return URI.create("/dragons/" + getName());
    }
}

File: webflux-demo-kotlin/src/main/kotlin/za/co/entelect/springforum/webfluxkotlin/dragons/Dragon.kt
package za.co.entelect.springforum.webfluxkotlin.dragons

import com.fasterxml.jackson.annotation.JsonIgnore
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

@Table("dragons")
data class Dragon(val name: String,
                  val location: String,
                  @Id @JsonIgnore val id: Int? = null)